dvc/stage.py

```diff
@@ -260,7 +260,10 @@ def remove_outs(self, ignore_remove=False):
         Used mainly for `dvc remove --outs`
         """
         for out in self.outs:
-            out.remove(ignore_remove=ignore_remove)
+            if out.persist:
+                self.repo.unprotect(out.path)
```

Comment:
This will lead to a bug in non-local out case. What we need to do is implement remote.unprotect and out.unprotect(that would call remote.unprotect). And also use remote.unprotect in repo.unprotect.